packages/vscode-extension/src/webview/components/NetworkRequestLog.tsx

```diff
@@ -0,0 +1,312 @@
+import { useState, useMemo, useEffect, useRef, useCallback } from "react";
+import { NetworkLog } from "../hooks/useNetworkTracker";
+import ResizableContainer from "./shared/ResizableContainer";
+import "./NetworkRequestLog.css";
+
+interface NetworkRequestLogProps {
+  networkLogs: NetworkLog[];
+  detailsWidth: number;
+  selectedNetworkLog: NetworkLog | null;
+  handleSelectedRequest: (id: string | null) => void;
+}
+
+const ROWS = ["Domain", "File", "Status", "Method", "Type", "Size", "Time"];
+const TABLE_RIGHT_PADDING = 0;
+const TABLE_CELL_MIN_WIDTH = 50;
+
+const NetworkRequestLog = ({
+  networkLogs,
+  detailsWidth,
+  handleSelectedRequest,
+  selectedNetworkLog,
+}: NetworkRequestLogProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [columnWidths, setColumnWidths] = useState<{ [key: string]: number }>(
+    ROWS.reduce(
+      (acc, title) => ({
+        ...acc,
+        [title]: 0,
+      }),
+      {}
+    )
+  );
+  const [lastDetailsWidth, setLastDetailsWidth] = useState(0);
+  const [initialRowWidth, setInitialRowWidth] = useState(0);
+
+  useEffect(() => {
```

Comment:
Shouldn't these be `layoutEffect`s, since you want to set UI element sizes?
https://react.dev/reference/react/useLayoutEffect